### PR TITLE
updated doc examples

### PR DESCRIPTION
**Pull request by Commandserver**
added `dpp::run_once<struct register_bot_commands>()` to the examples
added message content intent in bot ctor of the example whenever they're needed

### CODE REVIEW

File: docpages/03_example_programs.md [hunk 1]
```diff
@@ -314,7 +314,7 @@ int main(int argc, char const *argv[])
 	}
 
 	/* Setup the bot */
-	dpp::cluster bot("token");
+	dpp::cluster bot("token", dpp::i_default_intents | dpp::i_message_content); // Privileged intent required to recieve message content
```
Comment (braindigitalis) [inline docpages/03_example_programs.md:317]: receive

File: docpages/03_example_programs.md [hunk 2]
```diff
@@ -370,7 +370,7 @@ int main(int argc, char const *argv[])
 	 * You may use ffmpeg to encode songs to ogg opus:
 	 * ffmpeg -i /path/to/song -c:a libopus -ar 48000 -ac 2 -vn -b:a 96K /path/to/opus.ogg 
 	 */
-	dpp::cluster bot("token");
+	dpp::cluster bot("token", dpp::i_default_intents | dpp::i_message_content);
```
Comment (braindigitalis) [inline docpages/03_example_programs.md:373]: these examples need to be rewritten as slash commands. we shouldn't be giving examples which have message based commands except where absolutely necessary.

File: docpages/03_example_programs.md [hunk 3]
```diff
@@ -629,23 +629,23 @@ int main()
...
-			)
-		);
-
-		/* Register the command */
-		bot.global_command_create(newcommand);
+	    if (dpp::run_once<struct register_bot_commands>()) {
+
+		    dpp::slashcommand newcommand;
+		    /* Create a new global command on ready event */
+		    newcommand.set_name("blep", "Send a random adorable animal photo", bot.me.id)
```
Comment (braindigitalis) [inline docpages/03_example_programs.md:636]: this doesn't work.... the 3 parameter version is a constructor not the set_name method.

File: include/dpp/appcommand.h [hunk 4]
```diff
@@ -810,11 +810,11 @@ class DPP_EXPORT slashcommand : public managed {
 	/**
 	 * @brief Construct a new slashcommand object
 	 * 
-	 * @param _name Command name
-	 * @param _description Command description
-	 * @param _application_id Application id (usually the bot's user id)
+	 * @param name Command name
```
Comment (braindigitalis) [inline include/dpp/appcommand.h:813]: you can't change these, there's a reason for this (to silence doxygen warnings, and that you can't have a parameter name that shadows a name of a member variable

File: src/dpp/slashcommand.cpp [hunk 5]
```diff
@@ -33,10 +33,10 @@ using json = nlohmann::json;
 slashcommand::slashcommand() : managed(), application_id(0), type(ctxm_chat_input), default_permission(true), version(1) {
 }
 
-slashcommand::slashcommand(const std::string &_name, const std::string &_description, const dpp::snowflake _application_id) : slashcommand() {
-	set_name(_name);
-	set_description(_description);
-	set_application_id(_application_id);
+slashcommand::slashcommand(const std::string &name, const std::string &description, const dpp::snowflake application_id) : slashcommand() {
+	set_name(name);
```
Comment (braindigitalis) [inline src/dpp/slashcommand.cpp:37]: these shadow member variable names

**Review by braindigitalis — CHANGES_REQUESTED** [review]

File: docpages/03_example_programs.md (hunk 2 above)
Comment (Commandserver) [reply docpages/03_example_programs.md:373]: thats a quite complex change. it would rather change it in another pr :)

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**